ReSharper.FSharp/src/FSharp.Psi.Daemon/src/Analyzers/XmlDocAnalzyer.fs

```diff
@@ -11,3 +20,96 @@ type XmlDocAnalyzer() =
+    inherit ElementProblemAnalyzer<XmlDocBlock>()
+
+    let getNameAttribute (paramTag: IXmlTag) =
+        paramTag.Header.Attributes.FirstOrDefault(fun t -> t.AttributeName = "name")
+
+    let checkXmlTagIsNotClosed (tag: IXmlTag) =
+        let prevSibling = tag.PrevSibling
+        isNull prevSibling || not (prevSibling.GetText().EndsWith("[")) // [<attribute>]
+
+    let checkXmlHighlighting (highlighting: IHighlighting) =
+        match highlighting with
+        | :? XmlOnlyOneTagAllowedAtRootLevelHighlighting
+        | :? XmlNoRootTagDefinedHighlighting
+        | :? XmlTextIsNotAllowedAtRootHighlighting -> false
+        | :? XmlSyntaxErrorHighlighting as h when
+            h.ErrorElement.ErrorType = XmlSyntaxErrorType.INVALID_TAG_HEADER -> false
+        | :? XmlTagIsNotClosedHighlighting2 as h ->
+            checkXmlTagIsNotClosed h.Tags[0]
+        | :? XmlTagIsNotClosedHighlighting as h ->
+            checkXmlTagIsNotClosed h.Tag
+        | _ -> true
+
+    let checkXmlSyntax (xmlPsi: IDocCommentXmlPsi) (data: ElementProblemAnalyzerData) (consumer: IHighlightingConsumer) =
+        let daemonProcess = data.TryGetDaemonProcess()
+        if isNull daemonProcess then () else
+
+        let xmlFile = xmlPsi.XmlFile
+
+        let analyses =
+           xmlAnalysisManager.Providers
+           |> Seq.collect (fun provider -> provider.GetAnalyses(xmlFile, daemonProcess, data.SettingsStore))
+
+        let xmlConsumer = DefaultHighlightingConsumer(data.SourceFile)
+        let xmlAnalysisProcess = XmlAnalysisStageProcess(xmlFile, analyses, daemonProcess, xmlConsumer)
+
+        xmlAnalysisProcess.Execute(fun result ->
+        for highlighting in result.Highlightings do
+            if  not (checkXmlHighlighting highlighting.Highlighting) then () else
+
+            let warning = XmlDocCommentSyntaxWarning(highlighting.Highlighting, highlighting.Range)
+            consumer.AddHighlighting(warning, highlighting.Range))
+
+    let checkParameters (xmlPsi: IDocCommentXmlPsi) (xmlDocOwner: ITreeNode) (consumer: IHighlightingConsumer) =
+        let paramNodes = xmlPsi.GetParameterNodes(null)
+        //TODO: move to the platform?
+        let paramRefNodes = xmlPsi.XmlFile.Descendants<IXmlTag>().ToEnumerable()
+                            |> Seq.filter (fun x -> x.GetFullTagName() = "paramref")
+                            |> Seq.toArray
+
+        if paramNodes.Count = 0 && paramRefNodes.Length = 0 then () else
+
+        if xmlDocOwner :? IFSharpTypeDeclaration then () else
+        let parameters = FSharpParameterUtil.GetParametersGroupNames(xmlDocOwner)
+
+        let parameters = parameters |> Seq.collect id
+
+        for struct(name, parameter) in parameters do
+            if name = SharedImplUtil.MISSING_DECLARATION_NAME then () else
+
+            let parameterDocs = xmlPsi.GetParameterNodes(name)
+
+            if parameterDocs.Count = 0 then
+                consumer.AddHighlighting(XmlDocMissingParameterWarning(parameter))
```

Comment:
Let's make ranges of reported warnings cover names only in patterns and parameter signature type usages?